packages/apollo-server-express/src/ApolloServer.ts

```diff
@@ -183,3 +249,25 @@ export const registerServer = () => {
     'Please use server.applyMiddleware instead of registerServer. This warning will be removed in the next release',
   );
 };
+
+// XXX It would be great if there was a way to go through all properties of the parent to the Partial
+// to perform this check, but that does not exist yet, so we will need to check each one of the properties
+// of GuiOptions to see if there is anything set.
+function isPartialGui(gui: NonDynamicGuiConfig): gui is Partial<GuiOptions> {
+  return (
+    (<Partial<GuiOptions>>gui).playgroundThemeOptions !== undefined ||
+    (<Partial<GuiOptions>>gui).playgroundTabOptions !== undefined
+  );
+}
+
+// This is necessary to make TypeScript happy, since it cannot smart-cast types via conditional logic
+function isBoolean(gui: NonDynamicGuiConfig): gui is boolean {
+  return typeof gui === typeof true;
```

Comment:
I think this should be possible with an if statement like so:

```js
if(typeof gui === 'boolean') {
  ...
}
```

Reply:
I preferred doing it this way to leave out the hard-coded string. Is there a reason it's preferred to leave the hard-coded string instead of comparing it to a type you know?